### PR TITLE
FEATURE: allow for custom status codes on errors

### PR DESCRIPTION
**Pull request by treygriffith**
### Description
Allow users to pass custom status codes to errors for grpc

### Todo
- [x] Tests
- [x] Docs
- [x] Trello

### CODE REVIEW

**Comment by treygriffith** [comment]
[Close the subscribe-fill stream with a specific exception when the order has been cancelled by the user](https://trello.com/c/nw8y4Lmx/299-close-the-subscribe-fill-stream-with-a-specific-exception-when-the-order-has-been-cancelled-by-the-user)

File: src/grpc-method.js [hunk 1]
```diff
@@ -78,18 +78,19 @@ class GrpcMethod {
   /**
    * Format errors for consumption by external grpc clients
    *
-   * @param  {error} err Error to be formatted for public consumption
+   * @param  {error}  err        Error to be formatted for public consumption
+   * @param  {Number} statusCode grpc Status code
    * @return {GrpcError}
    */
-  grpcError (err) {
+  grpcError (err, statusCode = grpc.status.INTERNAL) {
```
Comment (dannypaz) [inline src/grpc-method.js]: Do you think it makes sense to expose grpc status codes in the library itself, or just allow ANY grpc status code?

File: src/grpc-method.js [hunk 2]
```diff
@@ -78,18 +78,19 @@ class GrpcMethod {
   /**
    * Format errors for consumption by external grpc clients
    *
-   * @param  {error} err Error to be formatted for public consumption
+   * @param  {error}  err        Error to be formatted for public consumption
+   * @param  {Number} statusCode grpc Status code
```
Comment (dannypaz) [inline src/grpc-method.js]: change Number to `grpc#status` or whatever the identifier is and make it optional with `[]`

File: src/grpc-method.js (hunk 1 above)
Comment (treygriffith) [reply src/grpc-method.js]: I think any grpc status codes - I don't see a ton of benefit in exposing them in this lib.

**Comment by dannypaz** [comment]
LGTM

**Review by dannypaz — APPROVED** [review]

**Pull request merged**